programs/client/Client.cpp

```diff
@@ -263,7 +259,20 @@ void Client::initialize(Poco::Util::Application & self)
     if (home_path_cstr)
         home_path = home_path_cstr;
 
-    configReadClient(config(), home_path);
+    std::optional<std::string> config_path;
+    if (config().has("config-file"))
+        config_path.emplace(config().getString("config-file"));
+    else
+        config_path = getClientConfigPath(home_path);
+    if (config_path.has_value())
+    {
+        ConfigProcessor config_processor(*config_path);
+        auto loaded_config = config_processor.loadConfig();
+        parseConnectionsCredentials(*loaded_config.configuration, config().getString("connection", ""));
```

Comment:
Looks like now if there is no client config file we will not call parseConnectionsCredentials and therefore not check client's options --host, etc while should

Reply:
There is a check below for this - https://github.com/ClickHouse/ClickHouse/pull/55949/files#diff-3db2d194d1cc74e956ad724b850fde753c12226216195e56a1f3125ef9c94da2R274

```
    else if (config().has("connection"))
        throw Exception(ErrorCodes::BAD_ARGUMENTS, "--connection was specified, but config does not exists");
```